ext/standard/array.c

```diff
@@ -4511,6 +4511,32 @@ PHP_FUNCTION(array_key_last)
 }
 /* }}} */
 
+PHP_FUNCTION(array_first)
+{
+	HashTable *array;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ARRAY_HT(array)
+	ZEND_PARSE_PARAMETERS_END();
+
+	ZEND_HASH_FOREACH_VAL(array, zval *zv) {
```

Comment:
TIL that I do not need to define the iteration variables before the `ZEND_HASH_FOREACH`, but can do so inline.